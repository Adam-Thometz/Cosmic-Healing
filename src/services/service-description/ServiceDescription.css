.ServiceDescription {
  display: flex;
  justify-content: space-evenly;
  padding: 5% 3%;
}

.reverse {
  flex-direction: row-reverse;
}

.ServiceDescription-description {
  width: 30%;
  text-align: left;
}

.ServiceDescription-description article {
  margin-bottom: 40px;
}

@media (max-width: 900px) {
  .ServiceDescription-description {
    width: 50%;
  }

  .ServiceDescription img {
    width: 50%;
  }
}

@media (max-width: 900px) {
  .ServiceDescription, .reverse {
    flex-direction: column;
    width: 95%;
    margin: 0 auto;
    align-items: center;
  }

  .ServiceDescription-description {
    width: 95%;
  }
}