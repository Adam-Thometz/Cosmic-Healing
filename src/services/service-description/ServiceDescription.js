import React from 'react';

import './ServiceDescription.css';

import Link from '../../_components/link/Link';
import SectionHeader from '../../_components/section-header/SectionHeader';

const ServiceDescription = ({ name, description, image, i }) => {
  return (
    <section className={`ServiceDescription${i % 2 ? ' reverse' : ''}`}>
      <img src={image} alt='' />
      <aside className='ServiceDescription-description'>
        <SectionHeader small text={name} />
        <article>{description}</article>
        <Link primary>Book a free session</Link>
      </aside>
    </section>
  );
};

export default ServiceDescription;