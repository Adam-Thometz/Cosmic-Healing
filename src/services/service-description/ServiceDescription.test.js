import React from "react";
import { render } from "@testing-library/react";
import ServiceDescription from './ServiceDescription';

describe('ServiceDescription component', () => {
  const component = <ServiceDescription
    name='Software Testing'
    description="I love testing so much"
    image='pic of guy just testing software'
    i={0}
  />;
  it('renders without crashing', () => {
    render(component);
  });

  it('matches the snapshot', () => {
    const { asFragment } = render(component);
    expect(asFragment()).toMatchSnapshot();
  });
});