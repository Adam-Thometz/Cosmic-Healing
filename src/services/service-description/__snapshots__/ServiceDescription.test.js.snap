// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ServiceDescription component matches the snapshot 1`] = `
<DocumentFragment>
  <section
    class="ServiceDescription"
  >
    <img
      alt=""
      src="pic of guy just testing software"
    />
    <aside
      class="ServiceDescription-description"
    >
      <h3
        class="SectionHeader"
      >
        Software Testing
      </h3>
      <article>
        I love testing so much
      </article>
      <a
        class="Link primary"
      >
        Book a free session
      </a>
    </aside>
  </section>
</DocumentFragment>
`;
